src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,49 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    /**
+     * Creates a type of task with a deadline
+     *
+     * @param description the content of the task
+     * @param by          time in yyyy-mm-dd
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDate.parse(by.trim());
+    }
+
+
+    public Deadline(String description, boolean isDone, String by) {
```

Comment:
It seems you missed out on this constructor when writing the JavaDoc for this class. Do remember to add it in when you're fine-tuning your code!